src/main/java/ch/heigvd/dai/Main.java

```diff
@@ -114,29 +124,39 @@ public static void main(String[] args) {
         "/delete/{id}",
         ctx -> {
           String id = ctx.pathParam("id");
-          if (id == null) {
-            ctx.status(400);
-            ctx.result("No id provided");
-          }
-
-          FileTransfer file = db.get(id);
+          FileTransfer file = getFileTransfer(id, db, ctx, true);
           if (file == null) {
-            ctx.status(404);
-            ctx.result("File not found");
-            return;
-          }
-
-          if (!file.checkAuthCode(ctx.queryParam("authCode"))) {
-            ctx.status(401);
-            ctx.result("Unauthorized");
             return;
           }
 
           db.remove(id);
+          file.getFile().delete();
           ctx.status(200);
           ctx.result("File deleted");
         });
 
     app.start(PORT);
   }
+
+  private static FileTransfer getFileTransfer(
+      String id, ConcurrentHashMap<String, FileTransfer> db, Context ctx, boolean authCheck) {
```

Comment:
No need for the method to be static, then the db parameter can be removed

Reply:
In fact it cannot be done because of the handler callback of javalin which is a static context and calling non static function into a static isn't possible. 
The only thing that can improve this would be to make a Handler class like in the users api from the course. What do you think ?